app/models/user.rb

```diff
@@ -2,13 +2,16 @@ class User < ApplicationRecord
 
   EMAIL_REGEXP = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i.freeze
 
-  after_destroy :ensure_an_admin_remains
   validates :name, presence: true, uniqueness: true
   has_secure_password
   validates :email, uniqueness: true, allow_nil: true
   validates :email, format: {
     with: EMAIL_REGEXP
   }
+  after_create :send_welcome_mail
```

Comment:
after_create will fire even before transaction is commited. What if transaction fails due to any reason?

Reply:
then welcome mail went but if I use .deliver_later than email is not send

Reply:
or i will make in after_commit and use on:create with it.